### ISSUE TITLE
Possible B020 false positive iterating over range()

### ISSUE THREAD

**Issue by MrkGrgsn** [PyCQA/flake8-bugbear#249]
This code:
```python
i = 0
for i in range(i, i + 10):
    print(i)
```
produces a `B020` warning. The loop isn't overriding the iterator so it appears to be a false positive to me.

Note: it's a dumb example but my actual use case involves an outer loop that sets `i` and the inner loop then increments `i` as per the example. Something like this:
```python
ranges = [(100, 150), (220, 223)]
start_value = ranges[0][0]
for range_start, range_end in ranges:
            i = max(start_value, range_start)
            for i in range(i, range_end):
                print(i)
```

**Comment by jpy-git** [comment]
even if this isn't actually overriding the loop iterable (like if you were updating a list whilst iterating over it) I would still argue it's not great practice to reuse that `i` variable for readability?

These examples do the same thing as your examples but it's clear to anyone reading the code that you aren't doing any sort of loop assignment:
```python
i = 0
for j in range(i, i + 10):
    print(j)
```

```python
ranges = [(100, 150), (220, 223)]
start_value = ranges[0][0]
for range_start, range_end in ranges:
    i = max(start_value, range_start)
    for j in range(i, range_end):
        print(j)
```

**Comment by cooperlees** [comment]
Agree with @jpy-git ... His examples produce much clearer and harder to misinterpret resultant code.

If you must do it, I think `#noqa: B020` is best here for this one ...

**Comment by MrkGrgsn** [comment]
Thanks for the feedback.  I can see your point, @jpy-git's examples do make the separation between inner and out loop clearer. I find it a subtle difference personally but happy to go with it. 👍
